sphinxlint.py

```diff
@@ -196,124 +197,206 @@ def is_in_a_table(error, line):
 role_missing_closing_backtick = re.compile(rf"({role_head}`[^`]+?)[^`]*$")
 
 
-def check_paragraph(paragraph_lno, paragraph):
-    if paragraph.count("|") > 4:
-        return  # we don't handle tables yet.
-    error = role_missing_closing_backtick.search(paragraph)
-    if error:
-        error_offset = paragraph[: error.start()].count("\n")
-        yield paragraph_lno + error_offset, f"role missing closing backtick: {error.group(0)!r}"
-    paragraph_without_roles = re.sub(normal_role, "", paragraph).replace("````", "")
-    for role in re.finditer("``.+?``(?!`).", paragraph_without_roles, flags=re.DOTALL):
-        if not re.match(end_string_suffix, role.group(0)[-1]):
-            error_offset = paragraph[: role.start()].count("\n")
-            yield paragraph_lno + error_offset, f"code sample missing surrogate space before plural: {role.group(0)!r}"
-
-
-@checker(".rst", severity=2)
-def check_suspicious_constructs_in_paragraphs(file, lines):
-    """Check for suspicious reST constructs at paragraph level."""
+@checker(".rst")
+def check_roles_missing_last_backtick(file, lines):
+    """Search for roles missing their closing backticks, like :fct:`foo."""
+    for paragraph_lno, paragraph in paragraphs(lines):
+        if paragraph.count("|") > 4:
+            return  # we don't handle tables yet.
+        error = role_missing_closing_backtick.search(paragraph)
+        if error:
+            error_offset = paragraph[: error.start()].count("\n")
+            yield paragraph_lno + error_offset, f"role missing closing backtick: {error.group(0)!r}"
+
+
+@checker(".rst")
+def check_inline_literal_glued_on_the_right(file, lines):
+    """Search for inline literals glued to plural, like ``items``s."""
```

Comment:
I don't think so, `literal` is a role name, while `inline literal` is a rst syntax: https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-literals

But for the name of a checker, it may be OK to use `literal`, for the sake of keeping them short.

Reply:
Indeed!  Do you know if there are other types of literals too? Multi-line literals?

Reply:
There's :

- [Literal Blocks](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#literal-blocks)
- [Indented Literal Blocks](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#indented-literal-blocks)
- [Quoted Literal Blocks](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#quoted-literal-blocks)

Reply:
> But for the name of a checker, it may be OK to use literal, for the sake of keeping them short.

Agreed, we can use `literal` for the checker name and `inline literal` in the docstring.